src/nixpacks/plan/generator.rs

```diff
@@ -231,16 +239,14 @@ impl<'a> NixpacksBuildPlanGenerator<'a> {
         Ok(static_assets)
     }
 
-    fn parse_procfile(&self, app: &App) -> Result<Option<String>> {
+    fn get_procfile_start_cmd(&self, app: &App) -> Result<Option<String>> {
         if app.includes_file("Procfile") {
-            let contents = app.read_file("Procfile")?;
-
-            // Better error handling
-            if contents.starts_with("web: ") {
-                return Ok(Some(contents.replace("web: ", "").trim().to_string()));
+            let procfile: Procfile = app.read_yaml("Procfile").unwrap_or_default();
+            if procfile.web.is_some() && procfile.worker.is_some() {
+                bail!("Procfile contains both a worker and a web process. Please specify one.");
+            } else {
+                Ok(procfile.web.or(procfile.worker))
```

Comment:
Maybe we should also warn the user if neither of the two are found? e.g. if they have something weird going on like
```yaml
web.1: npm run first
web.2: npm run second
```
In this scenario, the user doesn't technically have any of the supported process types. Would be cool to let them know why the build plan still runs the default commands.

Reply:
+1 that is a good idea